src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonService.kt

```diff
@@ -14,12 +15,16 @@ import java.time.LocalDateTime
 @Transactional
 class PrisonPersonService(
   private val bookingRepository: OffenderBookingRepository,
+  private val offenderRepository: OffenderRepository,
 ) {
-  fun getPhysicalAttributes(offenderNo: String) =
-    bookingRepository.findAllByOffenderNomsId(offenderNo)
-      .takeIf { it.isNotEmpty() }
-      ?.filterNot { it.physicalAttributes.isEmpty() }
-      ?.map {
+  fun getPhysicalAttributes(offenderNo: String): PrisonerPhysicalAttributesResponse {
+    if (!offenderRepository.existsByNomsId(offenderNo)) {
+      throw NotFoundException("No offender found for $offenderNo")
+    }
```

Comment:
We want to treat no booking the same as no physical attributes entered on bookings - this endpoint is just for a migration so zero is entirely expected. So we'll only return not found if the prisoner doesn't exist.